src/Popover.js

```diff
@@ -0,0 +1,234 @@
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import classnames from 'classnames'
+import {get} from './constants'
+import theme from './theme'
+import elementType from './utils/elementType'
+import Box from './Box'
+import BorderBox from './BorderBox'
+
+const Popover = styled(Box).attrs(({className, caret}) => {
+  return {
+    className: classnames(className, `caret-pos--${caret}`)
+  }
+})`
+  position: ${props => (props.relative ? 'relative' : 'absolute')};
+  z-index: 100;
+  display: ${props => (props.open ? 'block' : 'none')};
```

Comment:
Do you want consumers of this component to be able to override `display` or `z-index`? If so, you'll need to add `${LAYOUT}` to the end of these styles, so that custom values for anything in the `LAYOUT` group of props get applied after the styles you've authored here. If you do that, you might as well just do a `styled.div` here instead of extending `Box` since all `Box` is doing is implementing the `LAYOUT` and `COMMON` props. There's some discussion about this in [Derrick's Timeline PR](https://github.com/primer/components/pull/700#issuecomment-597901010).